### PR TITLE
NEWTAG-607 Data Tables

### PR DESCRIPTION
**Pull request by thpr**
NEWTAG-607 Lookup() function
NEWTAG-608 DATATABLE Campaign Token
NEWTAG-609 TABLE[x,y] format
NEWTAG-610 COLUMN[x] format

### CODE REVIEW

File: code/src/java/pcgen/cdom/format/table/TableFormatManager.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean equals(Object o)
+	{
+		if (o instanceof TableFormatManager)
+		{
+			TableFormatManager other = (TableFormatManager) o;
+			return lookupFormat.equals(other.lookupFormat)
+				&& resultFormat.equals(other.resultFormat);
```
Comment (cpmeister) [inline code/src/java/pcgen/cdom/format/table/TableFormatManager.java]: Unless TableFormatManager is final, this is a potentially erroneous equals() implementation

File: code/src/utest/pcgen/rules/persistence/TableLoaderTest.java [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+			//Yes
+		}
+		catch (IllegalArgumentException e)
+		{
+			//Yes
+		}
+		catch (IndexOutOfBoundsException e)
+		{
+			//Yes
+		}
```
Comment (cpmeister) [inline code/src/utest/pcgen/rules/persistence/TableLoaderTest.java]: You expect ALL of these errors?

Comment (thpr) [inline code/src/utest/pcgen/rules/persistence/TableLoaderTest.java]: Any one of is valid - all it needs to do is fail...

File: code/src/java/pcgen/cdom/format/table/TableFormatManager.java (hunk 1 above)
Comment (thpr) [inline code/src/java/pcgen/cdom/format/table/TableFormatManager.java]: I will change it to final, but I think the question is how much we want to worry about it... that is true almost universally across the entire java.* design, in how they originally implemented .equals()... they didn't use .getClass(), they just tested the local variables.  (The same is basically universally true across our code base.)

**Pull request merged**